iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/MergedBus.java

```diff
@@ -257,6 +257,22 @@ public Component getSynchronousComponent() {
         return buses.stream().flatMap(ConfiguredBus::getGeneratorStream);
     }
 
+    @Override
+    public Iterable<Battery> getBatteries() {
+        checkValidity();
+        List<Iterable<Battery>> iterables = new ArrayList<>(buses.size());
```

Comment:
It’s weird to reserve this capacity (todo: check how it’s done for other injections)

Reply:
It seems that every other component do the same allocation

Reply:
If it's really problematic, the other components should use an other allocation strategy.   
Maybe we should create a new issue for this